bokeh/models/annotations.py

```diff
@@ -107,6 +108,89 @@ class Legend(Annotation):
         Dict(String, List(Instance(GlyphRenderer))), lambda d: list(d.items())
     )
 
+class Arrow(Annotation):
+    """ Render an arrow as an annotation.
+
+    """
+
+    tail_x = NumberSpec(help="""
+    The x-coordinates to locate the arrow tails.
+    """)
+
+    tail_y = NumberSpec(help="""
+    The y-coordinates to locate the arrow tails.
+    """)
+
+    head_x = NumberSpec(help="""
+    The x-coordinates to locate the arrow heads.
+    """)
+
+    head_y = NumberSpec(help="""
+    The y-coordinates to locate the arrow heads.
+    """)
+
+    head_size = Float(default=25, help="""
+    The size, in pixels, of the arrow head.
+    """)
+
+    tail_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the head_x and head_y attributes. Interpreted as "data
+    space" units by default.
+    """)
+
+    head_units = Enum(SpatialUnits, default='data', help="""
+    The unit type for the head_x and head_y attributes. Interpreted as "data
+    space" units by default.
+    """)
+
+    tail_style = Enum(ArrowStyle, default=None, help="""
+    The style of the head's arrow tail.
```

Comment:
This sentence is a bit awkward. Maybe "A style for the arrow head at the head of the arrow. Now I actually wonder if head and tail for the coordinates is too confusing. What about start/end? Then: "A style for the arrow head at the start of the arrow"